ZON-4106: Be more strict when adding copyright information

knutgoetz opened the pull request:
Ticket: https://zeit-online.atlassian.net/browse/ZON-4106

Durch diesen PR soll die Datenqualität der Copyrightinformationen verbessert werden.

Beim Deployment nachtragen von `<company id="other">Andere</company>` in http://cms-backend.zeit.de:9000/cms/work/data/image-copyright-company.xml nicht vergessen.

zon-hoppe committed:
ZON-4106: Unify syntax

zon-hoppe committed:
ZON-4106: Get rid of copyright character

This will be rendered through friedbert directly.

zon-hoppe committed:
ZON-4106: Change widget order to make it more intuitive from user perspective

Since the value of the select box determines which input fields will be
rendered afterwards, it makes more sense to put the select above the
input fields.

zon-hoppe committed:
ZON-4106: Add js to dynamically render widgets

The select box value determines which input fields are rendered below.

In read only mode we will still render all possible fields, since the
DOM varies a lot from the write mode. While it is technically possible
to hide fields in ro mode as well, it requires lots of ugly DOM
navigation to accomplish this, so it is omitted as of now.

zon-hoppe committed:
ZON-4106: Add styles for image group copyrights

Commit:
MAINT: Delete whitespace vomit

Commit:
pep8

Commit:
ZON-4106: Fix tests

zon-hoppe committed:
ZON-4106: Revert widget order

This reverts most of 24e51d5cb59185ae7c07ba63666759a309c24cfe.

Unfortunately we cannot reorder fields in the model, because it will
mess with all existing content, not being mapped properly to the new
field order. Instead, all existing data would just be put over as is, in
the stored (read: previous) order over the new field order, which
results in fields having not their belonging data.

Moreover in some cases the code relies on a specific order, which we
would need to rewrite to work with both order models. This sounds like a
lot of headache!
See: metadata.ImageMetadata.copyrights

zon-hoppe committed:
ZON-4106: Revert widget order

This reverts most of 24e51d5cb59185ae7c07ba63666759a309c24cfe.

Unfortunately we cannot reorder fields in the model, because it will
mess with all existing content, not being mapped properly to the new
field order. Instead, all existing data would just be put over as is, in
the stored (read: previous) order over the new field order, which
results in fields having not their belonging data.

Moreover in some cases the code relies on a specific order, which we
would need to rewrite to work with both order models. This sounds like a
lot of headache!
See: metadata.ImageMetadata.copyrights

zon-hoppe committed:
ZON-4106: Rearrange widgets through styles instead of model

zon-hoppe committed:
ZON-4106: Rearrange widgets through styles instead of model

zon-hoppe committed:
ZON-4106: Apply PEP8

zon-hoppe committed:
ZON-4106: Apply PEP8

zon-hoppe committed:
ZON-4106: Restrict number of copyrights to exactly one

zon-hoppe committed:
ZON-4106: Restrict number of copyrights to exactly one

zon-hoppe committed:
ZON-4106: Apply styles for single copyrights

There ain't no multiple copyrights anymore.

zon-hoppe committed:
ZON-4106: Add copyright symbol to copyright fields

Commit:
Merge branch 'ZON-4106' of ssh://github.com/ZeitOnline/zeit.content.image into ZON-4106

Commit:
ZON-4106: Replace doctest with unit test for copyrights view

Commit:
FIX: Remove copyright.txt from doctest

Has been deleted

Commit:
ZON-4106: Refactor JS to work well with selenium tests

Commit:
ZON-4106: Change special case name in agency source.

This special case indicates, if a freetext copyright can be added

Commit:
Change CHANGES.txt

wosc commented on src/zeit/content/image/browser/resources/imagegroup.css:
```diff
@@ -0,0 +1,63 @@
+/* sequencewidget = checked out
+   sequenceWidget = checked in
+   ¯\_(ツ)_/¯   
+*/
```
OMG. Das vivi/formlib CSS ist echt gruselig. Mein Beileid.

wosc commented on src/zeit/content/image/interfaces.py:
```diff
@@ -58,8 +59,10 @@ class IImageMetadata(zope.interface.Interface):
+        default=((u'', None, None, None, False),),
         missing_value=(),
+        min_length=1,
+        max_length=1,
```
Sorry falls das eine total dämliche Frage ist, aber: Können wir dann nicht das `schema.Tuple` einfach komplett raushauen? Weil auf Anhieb seh ich nicht, wofür eine 1-elementige Liste gut sein soll. :)

knutgoetz replied:
Hatten wir auch überlegt. Ist ja jetzt etwas merkwürdig moddeliert. Der Grund warum wir das so gemacht haben, ist vor allem, dass wir uns dann BW-Compat Code sparen können.

wosc replied:
Hm... Im DAV ist das so oder so ein xmlpickle, d.h. das kann man dann behandeln, wie man will. Insofern ist glaub 1-2 Zeilen [hier in der Ecke](https://github.com/ZeitOnline/zeit.content.image/blob/master/src/zeit/content/image/metadata.py#L37) "wenn es eine Liste von Tupel ist, dann nimm nur das erste Element, danke" glaub ich das kleinere Übel, als im Friedbert etc.pp. überall ständig das ein(zig)e Element auswickeln zu müssen, oder?

wosc commented on src/zeit/content/image/browser/resources/form.js:
```diff
@@ -27,21 +27,48 @@
-        if (!$('fieldset.image-form').length) {
-            return;
-        }
+    var update_copyright_visibility = function() {
```
Yeah yeah. Good stuff. (Auch etwas gruselig, aber so ist glaub ich das vivi-JS-Leben zur Zeit.)

wosc reviewed (commented):
Super Sache, ich hab nur eine Frage, nämlich ob wir das Tupel nicht loswerden können, wenn es eh nur 1 Element hat.

wosc commented:
Übersetzungen hab ich direkt auf den zeit.locales master gepusht.

Commit:
Use simple combination

Commit:
pep8

Commit:
ZON-4106: Use only one combination of copyright information instead of multiple

In the past it was possible to add multiple combinations of copyright information (modelled as a tuple of tuples).

Commit:
ZON-4106: Adjust tests and templates to single copyright value

zon-hoppe committed:
ZON-4106: Adapt frontend code to new copyright structure

Unfortunately the sequenceWidget class is no longer available, so we
have to determine the state of the image group through a class from the
html body itself.

Commit:
ZON-4106: Add explicit bw compat test

Commit:
Merge branch 'ZON-4106' of ssh://github.com/ZeitOnline/zeit.content.image into ZON-4106

Commit:
ZON-4106: Use default value as missing value for copyrights

knutgoetz commented on src/zeit/content/image/interfaces.py:
```diff
@@ -56,30 +57,30 @@ class IImageMetadata(zope.interface.Interface):
-             zope.schema.Bool(
-                 title=_('set nofollow'),
-                required=False))))
+        missing_value=(u'', None, None, None, False))
```
Hier bin ich mir nicht sicher, ob das ne gute Idee ist. Motiviert durch diesen [Friedbert Code](https://github.com/ZeitOnline/zeit.web/blob/master/src/zeit/web/core/image.py#L771)

wosc replied:
Ich blick nicht so ganz durch, das weißt Du jetzt vmtl. am besten, aber nur mal ein ketzerischer Vorschlag: Wie wärs mit `None` als default/missing_value?

zon-hoppe committed:
ZON-4106: Add css class to determine ressource state

zon-hoppe committed:
ZON-4106: Manipulate DOM through js instead of relying on css alone

Unfortunately moving tablerows around with relative positioning is not
precisely predictable across different firefox versions on different
systems. Since even a single pixel difference in the checked-out state
will be noticeable, as the widgets photographer and copyrights freetext
overlap, we instead have to manipulate the DOM with javascript!

Since such a manipulation is even more messy in the checked-in state and
we do not rely on widget placement by pixel in this case, it is fine to
use relative positioning for the checked-in state.

zon-hoppe committed:
ZON-4106: Default copyrights to None

wosc reviewed (commented):
![copying](https://media.giphy.com/media/4aN6WVzkpYdCE/giphy.gif)
